### PR TITLE
feat: adjust OPA permissions

### PR DESCRIPTION
**Pull request by hsiliev**
Adjust OPA permissions

Solves PZ-2052

### CODE REVIEW

**Review by edgarvonk — APPROVED** [review]
No unit or integration tests?

I guess it's not so easy to test this stuff would be great to have some test coverage of these Rego policies somewhere..  There are ways to do that. E.g. https://www.openpolicyagent.org/docs/latest/policy-testing/

**Comment by hsiliev** [comment]
> No unit or integration tests?
> 
> I guess it's not so easy to test this stuff would be great to have some test coverage of these Rego policies somewhere..  There are ways to do that. E.g. https://www.openpolicyagent.org/docs/latest/policy-testing/

Plus these is https://github.com/Bisnode/opa-gradle-plugin

Let me see if I can add some tests with it

File: src/main/resources/policies/zaak-rechten.rego [hunk 1]
```diff
@@ -77,12 +97,184 @@ heropenen {
...
+    behandelaar.rol in user.rollen
+    zaaktype_allowed == true
+}
+
+default verlengen := false
+verlengen {
+    behandelaar.rol in user.rollen
+    zaaktype_allowed == true
+    zaak.open == true
+    # how do we check for reopened?
```
[the remark this answers is not in the document]
Comment (hsiliev) [reply src/main/resources/policies/zaak-rechten.rego:115]: Yep. I'm cleaning this up in the next PR where I expose the needed flags
I'll now have to add tests for these cases too :)

File: src/main/resources/policies/zaak-rechten.rego [hunk 2]
```diff
@@ -23,7 +24,26 @@ zaak_rechten := {
     "afbreken": afbreken,
     "heropenen": heropenen,
     "wijzigenZaakdata": wijzigenZaakdata,
-    "wijzigenDoorlooptijd": wijzigenDoorlooptijd
+    "wijzigenDoorlooptijd": wijzigenDoorlooptijd,
```
[the remark this answers is not in the document]
Comment (hsiliev) [reply src/main/resources/policies/zaak-rechten.rego:27]: Yeah. Lots of these indeed.

File: src/main/resources/policies/zaak-rechten.rego (hunk 1 above)
[the remark this answers is not in the document]
Comment (edgarvonk) [reply src/main/resources/policies/zaak-rechten.rego:115]: Not sure we need to add tests for all of them? Do these tests really add that enough value for this effort?

**Review by edgarvonk — APPROVED** [review]

File: src/main/resources/policies/zaak-rechten.rego (hunk 1 above)
[the remark this answers is not in the document]
Comment (hsiliev) [reply src/main/resources/policies/zaak-rechten.rego:115]: Well tests are already there and pretty fast.
Even caught a bug in one of the policies, so I would say its worth having them

We can always delete or disable them in the end

File: src/main/resources/policies/document-rechten.rego [hunk 3]
```diff
@@ -45,66 +46,107 @@ onvergrendeld_of_vergrendeld_door_user {
 
 default lezen := false
 lezen {
-    { behandelaar, coordinator, recordmanager }[_].rol in user.rollen
+    { behandelaar, coordinator, recordmanager, beheerder }[_].rol in user.rollen
```
Comment (bas-info-nl) [inline src/main/resources/policies/document-rechten.rego]: I thought rights should be stacked by adding roles to users, therefore only a `behandelaar` should have the read rights, or am I wrong?

**Review by bas-info-nl — APPROVED** [review]
Not sure if I understand correctly how this matches the roles matrix. But I see the code itself is "correct".

File: gradle/libs.versions.toml [hunk 4]
```diff
@@ -17,6 +17,8 @@ spotless-prettier-base = "3.2.5"
 spotless-prettier-organize-imports = "3.2.4"
 microprofile-rest-client = "3.0"
 jacoco = "0.8.12"
+opa-binary = "0.63.0"
```
[the remark this answers is not in the document]
Comment (hsiliev) [reply gradle/libs.versions.toml:20]: We have the opa gradle plugin (`opa-plugin`) and opa go binary (`opa-binary`) versions. Both are needed

File: src/main/resources/policies/zaak-rechten.rego (hunk 1 above)
[the remark this answers is not in the document]
Comment (edgarvonk) [reply src/main/resources/policies/zaak-rechten.rego:115]: ok, sounds good! nice addition :-)

ps: can we also measure this test coverage in Codecov? ;-)

[the remark this answers is not in the document]
Comment (hsiliev) [reply src/main/resources/policies/zaak-rechten.rego:115]: In theory yes. The coverage is generated in `build/reports/opa/opa-coverage.json`

Not sure how to provide this to Codecov though

**Pull request merged**